daemon/container.go

```diff
@@ -1275,19 +1277,31 @@ func (container *Container) initializeNetworking() error {
 	return container.buildHostnameAndHostsFiles(container.NetworkSettings.IPAddress)
 }
 
-// Make sure the config is compatible with the current kernel
-func (container *Container) verifyDaemonSettings() {
-	if container.hostConfig.Memory > 0 && !container.daemon.sysInfo.MemoryLimit {
-		logrus.Warnf("Your kernel does not support memory limit capabilities. Limitation discarded.")
-		container.hostConfig.Memory = 0
+// Make sure the config is compatible with the current kernel and daemon
+func (container *Container) verifyDaemonSettings() error {
+	hostConfig := container.hostConfig
+	daemon := container.daemon
+
+	if hostConfig.LxcConf.Len() > 0 && !strings.Contains(daemon.ExecutionDriver().Name(), "lxc") {
```

Comment:
@LK4D4 Yes, but it's related to daemon, if we created a container and passed this check, then we restart daemon with different exec driver, there would be a problem.
@jfrazelle Actually it's necessary, we could create a container, then change kernel and daemon setting and try to run this container. So daemon and kernel setting should always be checked when start containers.